File: src/test/java/de/fhopf/akka/HtmlParserTest.java
package de.fhopf.akka;

import org.htmlparser.Node;
import org.htmlparser.Parser;
import org.htmlparser.tags.LinkTag;
import org.htmlparser.util.ParserException;
import org.htmlparser.visitors.ObjectFindingVisitor;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

/**
 * Experiment with some of the html parser functionality.
 *
 * @author Florian Hopf, http://www.florian-hopf.de
 */
public class HtmlParserTest {

    @Test
    public void testLinkExtraction() throws ParserException {
        Parser parser = new Parser("http://synyx.de");
        ObjectFindingVisitor visitor = new ObjectFindingVisitor(LinkTag.class);
        parser.visitAllNodesWith(visitor);
        Node[] links = visitor.getTags();
        // TODO this could use some more meaningful assertions
        assertTrue(links.length > 0);
        for (int i = 0; i < links.length; i++) {
            LinkTag linkTag = (LinkTag) links[i];
            System.out.print("\"" + linkTag.getLinkText() + "\" => ");
            System.out.println(linkTag.getLink());
        }
    }
}

File: src/test/scala/de/fhopf/akka/HtmlParserTest.scala
package de.fhopf.akka

import org.htmlparser.Node
import org.htmlparser.Parser
import org.htmlparser.tags.LinkTag
import org.htmlparser.util.ParserException
import org.htmlparser.visitors.ObjectFindingVisitor
import org.junit.Test
import org.junit.Assert.assertTrue

/**
 * @author Florian Hopf, http://www.florian-hopf.de
 * @author Alfredo Serafini, http://www.seralf.it
 */
class HtmlParserTest {

  @Test
  def testLinkExtraction() = {
    val parser: Parser = new Parser("http://synyx.de")
    val visitor: ObjectFindingVisitor = new ObjectFindingVisitor(classOf[LinkTag])
    parser.visitAllNodesWith(visitor)
    val links: Array[Node] = visitor.getTags()
    assertTrue(links.length > 0)

    for (i <- 0 until links.length) {
      val linkTag = links { i }.asInstanceOf[LinkTag]
      print("\"" + linkTag.getLinkText() + "\" => ")
      println(linkTag.getLink())
    }

  }

}
